src/svtk/svtk/pesr/sr_test.py

```diff
@@ -9,59 +9,61 @@
 import numpy as np
 import scipy.stats as ss
 import pandas as pd
+import sys
 from .pesr_test import PESRTest, PESRTestRunner
 
 
 class SRTest(PESRTest):
-    def __init__(self, countfile, common=False, window=50, medians=None):
+    def __init__(self, countfile, common=False, window=50, ins_window=50, medians=None):
         self.countfile = countfile
         self.window = window
+        self.ins_window = ins_window
         self.common = common
 
         super().__init__(medians, common)
 
-    def test_record_rec(self, record, called, background):
-        # Test SR support at all coordinates within window of start/end
-        results = []
-        for coord in 'posA posB'.split():
-            result = self._test_coord(record, coord, called, background)
-            result['coord'] = coord
-            results.append(result)
-        results = pd.concat(results, ignore_index=True)
-
-        # Add test for sum of posA and posB
-        total = self._test_total(results)
-        results = pd.concat([results, total], ignore_index=True)
-
-        # Clean up columns
-        results['name'] = record.id
-        results['bg_frac'] = results.called / \
-            (results.background + results.called)
-        results['bg_frac'] = results.bg_frac.fillna(0)
-        cols = 'name coord pos log_pval called background bg_frac'.split()
-        return results[cols]
-
     def test_record(self, record, called, background):
         # Test SR support at all coordinates within window of start/end
         resultA = self._test_coord(
-            record, 'posA', record.chrom, called, background)
-        resultA['coord'] = 'posA'
+            record.pos, record.info['STRANDS'][0], record.chrom,
+            called, background, record.pos - self.window, record.pos + self.window)
 
         resultB = self._test_coord(
-            record, 'posB', record.info['CHR2'], called, background)
-        resultB['coord'] = 'posB'
-
-        if (record.chrom == record.info['CHR2']) and (int(resultA.pos) >= int(resultB.pos)):
-            rec = 1
-            while True:
-                result = self._test_coord_V2(
-                    record, 'posB', record.chrom, called, background, rec)
-                if int(result.pos) > int(resultA.pos.iloc[0]):
-                    result['coord'] = 'posB'
-                    resultB = result
-                    break
-                rec = rec + 1
+            record.stop, record.info['STRANDS'][1], record.info['CHR2'],
+            called, background, record.stop - self.window, record.stop + self.window)
+
+        posA = int(resultA.pos)
+        posB = int(resultB.pos)
+        log_pval_A = float(resultA.log_pval)
+        log_pval_B = float(resultB.log_pval)
+        if record.info['SVTYPE'] == 'INS':
+            if posA > posB + self.ins_window or posA < posB - self.ins_window:
```

Comment:
Does this mean that posA and posB must be within self.ins_window of each other for INS? What about INS with a deletion at the insertion site, or a sequence of microhomology > 50bp?

Reply:
Yes they could be larger, but I was attempting to match the intended behavior of the previous implementation which uses 50bp. Larger window sizes could increase fall positives, e.g. in messy regions you'd be more likely to find significant SR signals by chance. The microhomologous sequences are most frequently ~20bp but they can be larger. This is probably a point worth discussing so let's follow up.